### PR TITLE
`self.output_size()` (always `detail`-ed), introduce `self.num_coefficients`

### LINKED ISSUE #838: `precompute_size_scattering` should always return a tuple (`detail=False`)
Came up in conversation with @MuawizChaudhary 

The `detail` kwarg in `precompute_size_scattering` isn't really justified:
* it is `True` by default
* it creates type instability (`int` or `tuple`)
* getting the `False` version is trivial: it suffices to call `sum` on the output
* our object-oriented method `self.output_size`, has `detail=False` by default. I suggest removing that kwarg and simply summing across orders after calling `precompute_size_scattering`

Hence i propose to remove it. The only occasion where we need `self.output_size()` in the codebase is in the Keras frontend. 
Perhaps we can set up a helpful error message if people ever call the non-default: `self.output_size(detail=True)`, telling them to call `self.meta()` instead?

### PR DESCRIPTION
**Pull request by lostanlen**
Fixes #838 

* `precompute_size_scattering` now always returns a tuple
* same for `self.output_size()` in `Scattering1D`
* introduce read-only property `self.num_coefficients` to replace the old `self.output_size(detail=False)`
* fewer LOCs for `precompute_size_scattering`

This PR will conflict with #837 in the `if self.vectorize:` statements. #837  should take precedence

### CODE REVIEW

File: kymatio/scattering1d/frontend/base_frontend.py
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/base_frontend.py:122]: Wonderful

File: kymatio/scattering1d/frontend/numpy_frontend.py
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/numpy_frontend.py:49]: Could we not just do `size_scattering = self.num_coefficents`?

File: tests/scattering1d/test_torch_scattering1d.py
Comment (MuawizChaudhary) [inline tests/scattering1d/test_torch_scattering1d.py:280]: ?

File: tests/scattering1d/test_utils_scattering1d.py
Comment (MuawizChaudhary) [inline tests/scattering1d/test_utils_scattering1d.py:58]: why?

**Review by MuawizChaudhary — CHANGES_REQUESTED** [review]

File: tests/scattering1d/test_utils_scattering1d.py
Comment (lostanlen) [reply tests/scattering1d/test_utils_scattering1d.py:58]: To avoid this warning

```
/Users/user/kymatio/kymatio/scattering1d/filter_bank.py:309: UserWarning: Signal support is too small to avoid border effects
  warnings.warn('Signal support is too small to avoid border effects')
```

File: kymatio/scattering1d/frontend/numpy_frontend.py
Comment (lostanlen) [reply kymatio/scattering1d/frontend/numpy_frontend.py:49]: this code block will disappear with #837. i wouldn't worry about it

File: tests/scattering1d/test_torch_scattering1d.py
Comment (MuawizChaudhary) [inline tests/scattering1d/test_torch_scattering1d.py:269]: what do you think @janden

File: kymatio/scattering1d/frontend/numpy_frontend.py
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/numpy_frontend.py:51]: shouldnt this be removed?

File: kymatio/scattering1d/frontend/tensorflow_frontend.py
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/tensorflow_frontend.py:54]: ditto

File: kymatio/scattering1d/frontend/torch_frontend.py
Comment (MuawizChaudhary) [inline kymatio/scattering1d/frontend/torch_frontend.py:109]: ditto

**Review by MuawizChaudhary — CHANGES_REQUESTED** [review]

**Comment by lostanlen** [comment]
@janden your `output` is requested 😊

**Review by MuawizChaudhary — APPROVED** [review]
Most of my comments have been approved. @janden it would be nice to have your approval. Else I'll merge in a day ;)

**Comment by lostanlen** [comment]
actually, don't merge this. i'm realizing that we're switching from detail=False by default to detail=True by default. So we definitely need a deprecation cycle. I'll work on it

**Comment by MuawizChaudhary** [comment]
Wait what? I thought we discussed this and that this doesn't need a deprecation cycle? We are doing a sum on the output.

**Comment by lostanlen** [comment]
@MuawizChaudhary ah yes, sorry. i got confused between current default and intended default. So the implementation we want is `never-detail`, not `always-detail` right? But then why did you ask for a `num_coefficients` property?

**Pull request closed without merging**
